### DISCUSSION TITLE
Help setting up

### DISCUSSION THREAD

**Discussion by iKnowLessThanYou** [GiacomoLaw/Keylogger#144] [General]
See my username. I downloaded it and the folder has "klog_main.cpp" that I do not know how to open, and README which says this: "To change visibility of the window set the `#define` in line 9 to `visible` or `invisible`."

I don't know where I go to change any visibility? 

It also says, "Simply compile into an .exe, and then run. Visual Studio is good for this."

How?

**Comment by GiacomoLaw** [comment]
Read the source

**Reply by iKnowLessThanYou** [reply]
Can you help a guy with severe brain damage who struggles with technology? I googled where to find the "source," and it tells me about this Assets place. It has me download the same exact files I download elsewhere, and I don't know what to do. Please.

**Reply by GiacomoLaw** [reply]
https://github.com/GiacomoLaw/Keylogger/blob/ee4868265d9fa9d1972517d9ce9b8ffff286f2fd/windows/klog_main.cpp#L13C26-L13C26

**Reply by iKnowLessThanYou** [reply]
That looks like a lot of letters. Can you please explain this to me?

**Reply by chillmanstr8** [reply]
Same issue.  No idea what a .cpp file is, but I know it's not something that Visual Studio 2022 Community can build.  How is this compiled into an executable??

**Reply by chillmanstr8** [reply]
Also, the visible/invisible is on line 13 as of today.  Still no help with "compile with Visual Studio and you're done".. please shed some light on how to compile something that's not a solution file or project in VS.

**Reply by GiacomoLaw** [reply]
On line 13 in klog_main.cpp:

https://github.com/GiacomoLaw/Keylogger/blob/ee4868265d9fa9d1972517d9ce9b8ffff286f2fd/windows/klog_main.cpp#L13C26-L13C26

Change:

`#define visible // (visible / invisible)`

to

`#define invisible // (visible / invisible)`

To compile, try this:

* New project
* Visual C++ | Win32 | Win32 Project
* Console application
* Empty project
* Uncheck precompiled header
* Right click on source file and add the .cpp file
* Compile with F5

If that doesn't work, try this: https://learn.microsoft.com/en-us/cpp/windows/walkthrough-creating-a-standard-cpp-program-cpp?view=msvc-170

**Comment by Kingdamienjl** [comment]
so i got it all compiled with visual studio but it stops working after so long? its like i run it it captures strokes then if i dont make a keystroke for a while it closes itself? does that sound right? anyway i can keep it going and building that log.txt?
